_ml-commons-plugin/agents-tools/tools/rag-tool.md

```diff
@@ -0,0 +1,132 @@
+      "inference_model_id": "SNzSY40B_1JGmyB0WbfI",
+      "index": "my_test_data",
+      "embedding_field": "embedding",
+      "query_type": "neural",
+      "source_field": [
+        "text"
+      ],
+      "input": "${parameters.question}",
+      "prompt": "\n\nHuman:You are a professional data analyst. You will always answer question based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say don't know. \n\n Context:\n${parameters.output_field}\n\nHuman:${parameters.question}\n\nAssistant:"
+    }
+  }
+]
+}
+```
+{% include copy-curl.html %} 
+
+For parameter descriptions, see [Parameters](#parameters).
+
+OpenSearch responds with an agent ID:
+
+```json
+{
+  "agent_id": "9X7xWI0Bpc3sThaJdY9i"
+}
+```
+
+## Step 2: Run the agent
+
+Before you run the agent, make sure that you add the sample OpenSearch Dashboards `Sample web logs` dataset. To learn more, see [Adding sample data]({{site.url}}{{site.baseurl}}/dashboards/quickstart#adding-sample-data).
+
+Then, run the agent by sending the following request:
+
+```json
+POST /_plugins/_ml/agents/9X7xWI0Bpc3sThaJdY9i/_execute
+{
+  "parameters": {
+    "question": "what's the population increase of Seattle from 2021 to 2023"
+  }
+}
+```
+{% include copy-curl.html %} 
+
+OpenSearch performs vector search and returns the relevant documents:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "result": """{"_index":"my_test_data","_source":{"text":"Chart and table of population level and growth rate for the Seattle metro area from 1950 to 2023. United Nations population projections are also included through the year 2035.\\n
+          The current metro area population of Seattle in 2023 is 3,519,000, a 0.86% increase from 2022.\\n
+          The metro area population of Seattle in 2022 was 3,489,000, a 0.81% increase from 2021.\\n
+          The metro area population of Seattle in 2021 was 3,461,000, a 0.82% increase from 2020.\\n
+          The metro area population of Seattle in 2020 was 3,433,000, a 0.79% increase from 2019."},"_id":"6","_score":0.8173238}
+        {"_index":"my_test_data","_source":{"text":"Chart and table of population level and growth rate for the New York City metro area from 1950 to 2023. United Nations population projections are also included through the year 2035.\\n
+        The current metro area population of New York City in 2023 is 18,937,000, a 0.37% increase from 2022.\\n
+        The metro area population of New York City in 2022 was 18,867,000, a 0.23% increase from 2021.\\n
+        The metro area population of New York City in 2021 was 18,823,000, a 0.1% increase from 2020.\\n
+        The metro area population of New York City in 2020 was 18,804,000, a 0.01% decline from 2019."},"_id":"2","_score":0.6641471}
+        """
+        }
+      ]
+    }
+  ]
+}
+```
```

Comment:
we can also register RAGTool using conversational agent, shall we add instructions for conversation agent type as well?